bigquery/user_defined_functions/statistical-functions.sql

```diff
@@ -0,0 +1,43 @@
+  # Contains useful statistical functions that can be reused across multiple scheduled queries and views in BigQuery.
+  # This script must be run in BigQuery for these functions to become available, and re-run to overwrite the available functions with any new version.
+  #
+  # Test whether there is a probability greater than a specified confidence_level that two probabilities of a Boolean-valued outcome calculated from two independent samples are different
+  # e.g. test whether there is a >95% probability that the probabilities of a vacancy having a particular tag on a particular date and the same date in the previous year are different
```

Comment:
I'm confused about the idea of meta-probabilities.

Something like 'What are the chances that [the fact we had only 5% of vacancies tagged 'suitable for tigers' on 9th Sep 2020 but on 9th Sep 2019, 50% of vacancies were tagged 'suitable for tigers'] is a fluke vs. statistically significant?' ?

Reply:
Where '5%' and '50%' are frequencies being interpreted as probabilities?

Reply:
@david-mears-dfe yes, basically. Technically it's a comparison of two normal distributions (probability distributions of the proportion of 'successes') which are in turn approximations to the binomial distributions for the number of successes. The test statistic z bundles this all up into one calculation. Essentially we're asking the question 'what is the probability that these two probability distributions have the same mean'? And if it's < 5% we say that a finding that they *don't* have the same mean is statistically significant.